test/pynrn/test_py2nrnstring.py

```diff
@@ -0,0 +1,138 @@
+from neuron import h
+from neuron.expect_hocerr import expect_hocerr
+import sys
+
+uni = 'ab\xe0'
+
+def checking(s):
+  print("CHECKING: " + s)
+  pass
+
+def test_py2nrnstring():
+  print (uni)
+
+  # I don't think h.getstr() can be made to work from python
+  # so can't test unicode on stdin read by hoc.
+  h('strdef s')
+  h('s = "hello"')
+  checking ("h('getstr(s)')")
+  h('getstr(s)')
+  'goodbye'
+  assert(h.s == 'hello')
+
+  checking('h(uni)')
+  h(uni)
+
+  checking('h.printf("%s", uni)')
+  expect_hocerr(h.printf, ("%s", uni))
+
+  checking('hasattr(h, uni)')
+  expect_hocerr(hasattr, (h, uni))
+
+  checking('h.à = 1')
+  err = 0
+  try:
+    h.à = 1
+  except:
+    print((str(sys.exc_info()[0]) + ': ' + str(sys.exc_info()[1])))
+    err = 1
+  assert(err == 1)
+
+  checking('a = h.à')
+  err = 0
+  try:
+    a = h.à
+  except:
+    print((str(sys.exc_info()[0]) + ': ' + str(sys.exc_info()[1])))
+    err = 1
+  assert(err == 1)
+
+  checking('a = h.ref("à")')
+  err = 0
+  try:
+    a = h.ref("à")
+  except:
+    print((str(sys.exc_info()[0]) + ': ' + str(sys.exc_info()[1])))
+    err = 1
+  assert(err == 1)
+
+  ns = h.NetStim()
+  #nonsense but it does test the error unicode error message
+  checking('h.setpointer(ns._ref_start, "à", ns)')
+  expect_hocerr(h.setpointer, (ns._ref_start, "à", ns))
+  del ns
+
+  '''
+  # No error for these (unless cell is involved)!
+  checking('h.Section(name=uni)')
+  err = 0
+  try:
+    h.Section(name=uni)
+  except:
+    print((str(sys.exc_info()[0]) + ': ' + str(sys.exc_info()[1])))
+    err = 1
+  assert(err == 1)
+
+  checking('h.Section(uni)')
+  expect_hocerr(h.Section, (uni,))
+  '''
```

Comment:
If that's not supposed to raise an error message, shouldn't that be checked instead of having them check be commented out?